web/src/copy_and_paste.js

```diff
@@ -330,6 +330,34 @@ export function paste_handler_converter(paste_html) {
     return markdown_text;
 }
 
+function is_safe_url_paste_target($textarea) {
+    const range = $textarea.range();
+
+    if (!range.text) {
+        // No range is selected
+        return false;
+    }
+
+    if (isUrl(range.text.trim())) {
+        // Don't engage our URL paste logic over existing URLs
+        return false;
+    }
+
+    // To avoid doubling up on Markdown link syntax, we want to look
+    // 2 characters back from the original range start for `](`
+    const possible_markdown_link_markers = $textarea.range(range.start - 2, range.start).range();
```

Comment:
Does this behave correctly if `range.start < 2`?

Reply:
It does. If a negative number is passed in, the returned range believes itself to be starting at 0. So unlike other methods that would go to the end of something and work backwards, this has a hard start at 0.

Reply:
I’m seeing it count negative indexes backward from the end, and this is confirmed by the [source](https://github.com/acdvorak/jquery.caret/blob/v1.5.2/src/jquery.caret.js#L38-L41).

Reply:
Got it. I was going by the logging the returned range, but that's probably not the complete picture. I'll update with a condition for when the starting range is less than two.

Reply:
I suspect I was seeing that 0 value for start and end from the returned range because [setSelectionRange](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange) sets the value of `selectionStart` to the value of `selectionEnd` when the start value is less than end. And that has the effect of clearing the selection entirely, which is why I was seeing 0 start and end values in the negative case.